### PR TITLE
Use sticky scheduling.

### LINKED ISSUE #150: Use operator rs `create_missing_pods` with Sticky Scheduler

### PR DESCRIPTION
**Pull request by razvan**
## Description

Fixes #150 

Depends on: https://github.com/stackabletech/operator-rs/pull/222

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] git dependencies don't  references branches

### CODE REVIEW

File: rust/operator/src/lib.rs [hunk 1]
```diff
@@ -207,58 +210,66 @@ impl SparkState {
...
-                    }
+                        history.save(&self.context.resource).await?;
 
-                    // we do this here to make sure pods of each role are started after each other
-                    // roles start in rolling fashion
-                    // pods of each role group start in parallel
-                    // master > worker > history server
-                    if changes_applied {
                         return Ok(ReconcileFunctionAction::Requeue(Duration::from_secs(10)));
                     }
```
Comment (maltesander) [inline rust/operator/src/lib.rs:274]: Is that still a rolling fashion (e.g. all masters, then all workers...)?

**Review by maltesander — DISMISSED** [review]
Just one question, looks good otherwise.

File: rust/operator/src/lib.rs (hunk 1 above)
Comment (razvan) [reply rust/operator/src/lib.rs:277]: The order in which roles are deployed is determined by `SparkRole::iter()` and that does master, worker and history.

**Review by maltesander — CHANGES_REQUESTED** [review]
Just minor things, looks good otherwise.

File: rust/operator/src/lib.rs [hunk 2]
```diff
@@ -293,19 +306,19 @@ impl SparkState {
...
 
         let mut cm_labels = get_recommended_labels(
             &self.context.resource,
-            APP_NAME,
+            pod_id.app(),
             &self.context.resource.spec.version.to_string(),
-            role,
-            group,
+            pod_id.role(),
+            pod_id.group(),
```
[the remark this answers is not in the document]
Comment (razvan) [reply rust/operator/src/lib.rs:323]: You are right, but the `get_recommended_labels` function is in `operator-rs` and that PR  is already pretty large. Maybe in the next iteration.

File: rust/operator/src/lib.rs [hunk 3]
```diff
@@ -386,16 +399,15 @@ impl SparkState {
     ///
     async fn create_pod(
         &self,
+        pod_id: &PodIdentity,
         role: &SparkRole,
```
[the remark this answers is not in the document]
Comment (razvan) [reply rust/operator/src/lib.rs:405]: Unfortunately yes. It is needed for several things in this function.

File: rust/operator/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator/src/lib.rs:328]: yeah i confused it with required_pod_labels.

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request merged**